build.gradle

```diff
@@ -563,6 +579,48 @@ project('test:system') {
 
         onlyIf { dockerRegistryUrl && repoUrl }
     }
+
+    task execShellScript(type: Exec) {
+        commandLine "./preTestScript.sh"
+        args "$CLUSTER_NAME", "$MASTER", "$NUM_SLAVES"
+    }
+
+    //This is  used to invoke docker based systemTests
+    task startSystemTestsWithDocker(type: Test) {
+
+            dependsOn 'testJarForDocker', 'execShellScript'
+
+        ext.dockerRegistryUrl = project.hasProperty("dockerRegistryUrl") ? project.dockerRegistryUrl : ""
+
+        description 'Used to invoke docker based system tests, example usage: gradle startSystemTests -DmasterIP=xx.xx.xx.xx'
+        testClassesDir = sourceSets.test.output.classesDir
+        classpath = sourceSets.test.runtimeClasspath
+
+        systemProperty "execType", System.getProperty("execType")
+        systemProperty "masterIP", System.getProperty("masterIP")
+        systemProperty "imageVersion", System.getProperty("imageVersion")
+        systemProperty "skipServiceInstallation", System.getProperty("skipServiceInstallation")
```

Comment:
We do not require this property for docker based run.

Reply:
It is needed, but we dont provide anything while running gradle task, because am setting the default to `false` down here, and System.getProperty() is defined here for the same.